app/utils/nonsipp/CheckStatusUtils.scala

```diff
@@ -0,0 +1,175 @@
+   * @param recordIndex the index of the record being checked
+   * @return true if the record requires checking, else false
+   */
+  def checkLandOrPropertyRecord(
+    userAnswers: UserAnswers,
+    srn: Srn,
+    recordIndex: Max5000
+  ): Boolean = {
+    val prePopClearedAnswersMissing: Boolean = (
+      userAnswers.get(IsLandOrPropertyResidentialPage(srn, recordIndex)),
+      userAnswers.get(IsLandPropertyLeasedPage(srn, recordIndex)),
+      userAnswers.get(LandOrPropertyLeaseDetailsPage(srn, recordIndex)),
+      userAnswers.get(IsLesseeConnectedPartyPage(srn, recordIndex)),
+      userAnswers.get(LandOrPropertyTotalIncomePage(srn, recordIndex))
+    ) match {
+      case (None, None, None, None, None) => true
+      case (_, _, _, _, _) => false
+    }
+
+    lazy val otherAnswersPresent: Boolean = (
+      userAnswers.get(LandPropertyInUKPage(srn, recordIndex)),
+      userAnswers.get(LandOrPropertyChosenAddressPage(srn, recordIndex)),
+      userAnswers.get(LandRegistryTitleNumberPage(srn, recordIndex)),
+      userAnswers.get(WhyDoesSchemeHoldLandPropertyPage(srn, recordIndex)),
+      userAnswers.get(LandOrPropertyWhenDidSchemeAcquirePage(srn, recordIndex)),
+      // IdentityType pages are handled by the identitySubjectAnswersPresent method
+      userAnswers.get(LandOrPropertySellerConnectedPartyPage(srn, recordIndex)),
+      userAnswers.get(LandOrPropertyTotalCostPage(srn, recordIndex)),
+      userAnswers.get(LandPropertyIndependentValuationPage(srn, recordIndex))
+    ) match {
+      // Acquisition
+      case (Some(_), Some(_), Some(_), Some(Acquisition), Some(_), Some(_), Some(_), Some(_)) =>
+        userAnswers.get(IdentityTypePage(srn, recordIndex, LandOrPropertySeller)) match {
+          case None => false
+          case Some(identityType) =>
+            identitySubjectAnswersPresent(userAnswers, srn, recordIndex, LandOrPropertySeller, identityType)
+        }
+      // Contribution
+      case (Some(_), Some(_), Some(_), Some(Contribution), Some(_), None, Some(_), Some(_)) =>
+        true
+      // Transfer
+      case (Some(_), Some(_), Some(_), Some(Transfer), None, None, Some(_), None) =>
+        true
+      case (_, _, _, _, _, _, _, _) =>
+        false
+    }
+
+    prePopClearedAnswersMissing && otherAnswersPresent
+  }
+
+  /**
+   * This method determines whether or not all answers are present for a given IdentityType & IdentitySubject.
+   * @param userAnswers the answers provided by the user
+   * @param srn the Scheme Reference Number, used for the .get calls
+   * @param recordIndex the index of the record being checked
+   * @param identitySubject relates to the section this is called from: Loans, Land or Property, Shares, or Other Assets
+   * @param identityType relates to the seller involved: Individual, UKCompany, UKPartnership, or Other
+   * @return true if all answers are present, else false
+   */
+  def identitySubjectAnswersPresent(
```

Comment:
very minor: could you mark functions that will not be used outside of this class as private

Reply:
That's a good shout, but this method's also used in CheckStatusUtilsSpec, so I can't set it as private because then it'll be inaccessible for those tests.

Reply:
if you're dead set on testing this function in isolation, you can set it to package private

Reply:
Okay, I'll take what you've both said onboard and set it as private, then restructure the tests accordingly. I'll push that as a separate commit, and if the consensus is that it's better that way then we can keep that commit, and if not, then we can revert that commit.

Reply:
No it's alright @westwater , I don't feel particularly strongly about it, I just thought it made sense to organise the tests by each method they're testing.